packages/myst-transforms/src/targets.ts

```diff
@@ -23,8 +23,11 @@ import { normalizeLabel, toText } from 'myst-common';
 export function mystTargetsTransform(tree: Root) {
   visit(tree, 'mystTarget', (node: Target, index: number, parent: Parent) => {
     // TODO: have multiple targets and collect the labels
-    const nextNode = findAfter(parent, index) as any;
+    const child = findAfter(parent, index) as any;
     const normalized = normalizeLabel(node.label);
+    // A mystDirective will be removed in a later step
+    // Put the target on the first child if it exists
+    const nextNode = child?.type === 'mystDirective' ? child.children?.[0] : child;
```

Comment:
This makes sense - calls slightly into question the order these transforms take place, i.e. why don't we remove `mystDirectives` before propagating targets?

But this makes sense to get things working now.

Reply:
Fixed up taking this suggestion.